libs/jsonschema/schema.go

```diff
@@ -87,6 +88,43 @@ type Schema struct {
 	Deprecated bool `json:"deprecated,omitempty"`
 }
 
+func anyToSchema(a any) (*Schema, error) {
+	b, err := json.Marshal(a)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal any to bytes: %w", err)
+	}
+
+	res := &Schema{}
+	err = json.Unmarshal(b, res)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal bytes to JSON schema: %w", err)
+	}
+
+	return res, nil
+}
+
+func (s *Schema) GetDefinition(p string) (*Schema, error) {
+	parts := strings.Split(p, "/")
+	if parts[0] != "#" || parts[1] != "$defs" {
+		return nil, fmt.Errorf("invalid reference %q. References must start with #/$defs/", p)
+	}
+
+	curr := s.Definitions
+	var ok bool
+	for i, part := range parts[2:] {
+		if i == len(parts)-3 {
+			return anyToSchema(curr[part])
+		}
+
+		curr, ok = curr[part].(map[string]any)
+		if !ok {
+			return nil, fmt.Errorf("invalid reference %q. Failed to resolve reference at %q", p, part)
+		}
+	}
+
+	return nil, fmt.Errorf("invalid reference %q. Expected more than 2 tokens", p)
```

Comment:
I recommend doing the bounds check before the loop.

Then you can remove the inner if condition and let the loop terminate.

Reply:
Good point, changed.